lib/mongo/client.rb

```diff
@@ -1112,8 +1112,26 @@ def cluster_modifying?(new_options)
     # but does not check for interactions between combinations of options.
     def validate_new_options!(opts = Options::Redacted.new)
       return Options::Redacted.new unless opts
+      if opts[:read_concern]
+        # Raise an error for non user-settable options
+        if opts[:read_concern][:after_cluster_time]
+          raise Mongo::Error::InvalidReadConcern.new(
+            'The after_cluster_time read_concern option cannot be specified by the user'
+          )
+        end
+
+        given_keys = opts[:read_concern].keys
```

Comment:
This is relying on the strings being keys which might change due to Ruby 3 work, can there be an additional `.map(&:to_s)` to stringify them?

Reply:
Good point!